packages/transformers/js/src/visitors/dependencies.js

```diff
@@ -525,3 +538,28 @@ function objectExpressionNodeToJSONObject(
 
   return object;
 }
+
+function normalize(asset: MutableAsset, templateLiteral: TemplateLiteral) {
+  if (templateLiteral.expressions.length) {
+    let loc = convertBabelLoc(templateLiteral.loc);
+    throw new ThrowableDiagnostic({
+      diagnostic: {
+        message: 'Expressions are not allowed in the require() calls.',
+        origin: '@parcel/transformer-js',
+        ...(loc && {
+          codeFrame: {
+            codeHighlights: [{start: loc.start, end: loc.end}],
+          },
+        }),
+        filePath: asset.filePath,
+      },
+    });
```

Comment:
I'm not really sure if we should throw an error here. It makes sense for use code, but not really if some node_module does this and that code is actually never invoked at runtime.

Maybe it should only throw if `asset.isSource` and otherwise do `logger.warn`.